### PR TITLE
Use improved Thing methods for PostNavigator ShowLink

### LINKED ISSUE #3488: PageNavigator ShowLink shows empty () on ads pages
Find an ad post, eg from the spotlight,, and look in the top right corner

### PR DESCRIPTION
**Pull request by andytuba**
Fixes #3488

### CODE REVIEW

File: lib/utils/Thing.js [hunk 1]
```diff
@@ -218,8 +245,28 @@ export default class Thing {
...
+		}
+		return `/domain/${this.getPostDomain()}/`;
+	}
+
 	getPostDomainLink() {
-		return this.thing.querySelector('.domain > a');
+		return this.thing.querySelector('.domain a');
+	}
+
+	getPostDomainText() {
```
Comment (andytuba) [inline lib/utils/Thing.js]: I feel like there are some redundancies here that could be consolidated.

Comment (erikdesjardins) [inline lib/utils/Thing.js]: You could have stuff like:

```js
_getDomainFromAttrs() {
	return this.thing.getAttribute('data-domain');
}
```

but I don't think it's a huge deal.

Comment (andytuba) [inline lib/utils/Thing.js]: Yeah, I think I'll just stick with some minor duplication. Even if that logic changes a little, it's all tidily grouped together.

**Comment by andytuba** [comment]
:haircut_man:

**Pull request merged**